pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/io/params/RecommenderConstants.java

```diff
@@ -80,6 +81,13 @@
     public static final String ERROR_INVALID_QUERY = "Error: query not able to parse, skipped";
   }
 
+  public static class RealtimeProvisioningRule {
```

Comment:
Javadoc please. I see classes for other rules here don't have javadocs. We can probably add those docs later. But for new code, let's document as much as possible.

Reply:
Here the constants are defined for RealtimeProvisiningRule. I'm not sure adding comments makes it more clear than that. Some experts believe that we should not put comments as much as we can and instead use descriptive names unless there's no way around it. They believe that in code refactoring comments don't get proper attention and easily become outdated.
Anyways, if you still think it's better to write something there, I can simply add it.